ShoppingList/src/main/java/edu/agh/eaiib/CommandParser.java

```diff
@@ -14,16 +15,22 @@ public void parse(String input) {
             System.out.println("add <number> <productName> to <listName> - adds a product to the list, if the list doesn't exist, the system makes it with the specified entry");
             System.out.println("create <listName> - creates a list with the specified name and the user as the creator of the list");
             System.out.println("buy <productName> in <listName> - marks the product as bought");
-        } else {
-            if (input.matches("login [A-Za-z0-9]+")) {
+        } else if (input.matches("login [A-Za-z0-9]+")) {
 
-            } else if (input.matches("add [0-9]+ [A-Za-z0-9]+ to [A-Za-z0-9]+")) {
+        } else if (input.matches("add [0-9]+ [A-Za-z0-9]+")) {
+            parseAdd(input);
+        } else if (input.matches("create [A-Za-z0-9]+")) {
 
-            } else if (input.matches("create [A-Za-z0-9]+")) {
+        } else if (input.matches("buy [A-Za-z0-9]+ in [A-Za-z0-9]+")) {
```

Comment:
I have in mind nice refactoring using Predicate class. But maybe not this time 😄 
And miss Kotlin `when` expression (and Kotlin in general 😢 )